internal/controller/progressive/progressive.go

```diff
@@ -274,10 +269,8 @@ func processUpgradingChild(
 
 		// if so, mark the existing one for garbage collection and then create a new upgrading one
 		if needsUpdating {
-			if liveRolloutObject.GetPromotedChildStatus() == nil {
+			if rolloutObject.GetPromotedChildStatus() == nil {
 				rolloutObject.SetPromotedChildStatus(&apiv1.PromotedChildStatus{Name: existingPromotedChildDef.GetName()})
-			} else {
-				rolloutObject.SetPromotedChildStatus(liveRolloutObject.GetPromotedChildStatus().DeepCopy())
```

Comment:
no need for this copy - we already copied the liveRolloutObject's entire Progressive Status into here